Dodge.js

```diff
@@ -0,0 +1,186 @@
+      console.log(i)
+      cb(i)
+  }, seconds * 1000);
+}
+
+function DeleteEntity(i){
+delete(ents[i])
+}
+
+function Explode(i){
+  //spawns shrapnel coming from location
+  InitialShrapnel(ents[i].core.x,ents[i].core.y)
+  //delete the original projectile
+  DeleteEntity(i)
+}
+
+function Gameloop(entities, { touches }, )  {
+  //setTimeout is NOT what I want?
+  for (let id in entities) {
+    entities[id].core.update()
+  }
+  
+return entities
+}
+
+function CreateBrick(props){
+  return (
+    <View
+      style={{
+        position: 'absolute',
+        width: props.core.width,
+        height: props.core.height,
+        left: props.core.x,
+        top: props.core.y,
+        backgroundColor: props.core.backgroundColor,
+      }}
+    />
+  );
+}
+
+function CreateCircle(props){
+  return (
+
+    <View
+      style={{
+        position: 'absolute',
+        borderRadius: props.core.borderRadius,
+        width: props.core.width,
+        height: props.core.height,
+        left: props.core.x,
+        top: props.core.y,
+        backgroundColor: props.core.backgroundColor,
+      }}
+    />
+  );
+}
+
+function CreateGrenade(props){
+  return (
+    <View
+      style={{
+        position: 'absolute',
+        borderRadius: props.core.borderRadius,
+        width: props.core.width,
+        height: props.core.height,
+        left: props.core.x,
+        top: props.core.y,
+        backgroundColor: props.core.backgroundColor,
+      }}
+    />
+  );
+}
+
+  
+
+var ents = {knight: { core: new Knight(), renderer: <CreateBrick />},};
+  //Knight and Brick run every time the game updates that is not good
+
+  function InitialCircle() {
+    let random = Math.floor(Math.random()*500) + 75
+    ents[index] = { core: new Circle(random, 15), renderer: <CreateCircle /> };
+    index++
+  }
+
+  function InitialBrick(){
+    let random = Math.floor(Math.random()*500) + 75
+    ents[index] = { core: new Brick(random, 15), renderer: <CreateBrick />  };
+    index++
+  }
+
+  function InitialGrenade(){
+    let random = Math.floor(Math.random()*500) + 75
+    ents[index] = { core: new Grenade(random, 15), renderer: <CreateCircle />,}
+    //console.log(index)
+    Timer(Explode,index, 4)
+    index++
+  }
+
+  function InitialShrapnel(x , y){
+    let dY = 0
+    for(let dX= -1; dX < 2; dX++){
+//dX and dY are the direction the shrapnel will head in
+        if(dX == 0 ){
+          ents[index] = { core: new Shrapnel(x,y,dX,-1), renderer: <CreateCircle/> }
+          Timer(DeleteEntity, index, 2)
+          index++
+          ents[index] = { core: new Shrapnel(x,y,dX,1), renderer: <CreateCircle/> }
+          Timer(DeleteEntity, index, 2)
+          index++
+        }
+ else {
+  ents[index] = { core: new Shrapnel(x,y,dX,0), renderer: <CreateCircle/> }
+  Timer(DeleteEntity, index, 3)
+  index++
+ }
+}
+}
+
+function TimeUp(navigation){
+  {navigation.navigate("Results", {counter: counter})}
+
+}
+
+function reset(){
```

Comment:
Example comment:
// Remove all entities from the screen.

Consider a more descriptive name, e.g. clearEntities()